echo_client.py

```diff
@@ -0,0 +1,28 @@
+# -*- coding: utf-8 -*-
+import socket
+import sys
+
+def client(message):
+    message = message.encode('utf-8')
```

Comment:
If the message passed is already a byte string, this will cause problems in Python 2.  Python will see that you want to encode a unicode object, note that it has a byte string, and attempt to convert the byte string into a unicode object.  Since you don't have control over that conversion, you'll get the `sys.defaultencoding` which is likely ASCII, and if your byte string had extended code points in it, it would break.